### DISCUSSION TITLE
unable to get lsp to work for python code in qmd file...again...

### DISCUSSION THREAD

**Discussion by hopezh** [quarto-dev/quarto-nvim#98] [Q&A] [answered]
For example, I got a warning, rather than a `hover` info when use `K` on the variable `r`, as shown below. 
<img width="1244" alt="Screenshot 2023-10-30 at 20 21 45" src="https://github.com/quarto-dev/quarto-nvim/assets/23435810/ced5e3fb-db0a-465e-a0a1-0647979788b7">

...and this is what I should see (using [lazyvim-starter-for-quarto](https://github.com/jmbuhr/lazyvim-starter-for-quarto)):
<img width="1244" alt="Screenshot 2023-10-30 at 20 30 27" src="https://github.com/quarto-dev/quarto-nvim/assets/23435810/5d8b3ec7-1175-4d97-83b2-e3fa37ca313d">

I checked and it seems I do have lsp for python and marksman attached:
<img width="1244" alt="Screenshot 2023-10-30 at 20 22 04" src="https://github.com/quarto-dev/quarto-nvim/assets/23435810/b4317d41-b302-44b8-ad35-427874dff761">

My I ask what I'm missing here, or incorrectly configured, that is causing the missing of lsp function? 

My config file: 
- [oatLazyVim](https://github.com/hopezh/oatLazyVim)
- [lazy.lua](https://github.com/quarto-dev/quarto-nvim/discussions/new?category=q-a)
- [quarto.lua](https://github.com/hopezh/oatLazyVim/blob/main/lua/plugins/quarto.lua)
- [nvim_lspconfig](https://github.com/hopezh/oatLazyVim/blob/main/lua/plugins/nvim_lspconfig.lua)
- [nvim_treesitter.lua](https://github.com/hopezh/oatLazyVim/blob/main/lua/plugins/nvim_treesitter.lua)

**Comment by jmbuhr** [comment]
Are you making sure that the extra language servers that attach to the quarto buffer don't overwrite the `K` keymap from their otter.nvim version we us in quarto-nvim to the regular lsp hover?
If your config is based in LazyVim, it might be that the order in which your plugins are configured matters due to https://github.com/quarto-dev/quarto-nvim/blob/14a80ffab90567e360eb97b19039c4c16913ad79/ftplugin/quarto.lua#L22-L36

If your config uses your own `on_attach`, see how I use two different on_attach functions depending on the filetype to make not overwrite quarto keymaps:

https://github.com/jmbuhr/quarto-nvim-kickstarter/blob/b82936d4391aeca65689b91f0687476fe1933807/lua/plugins/quarto.lua#L133-L176

**Comment by hopezh** [comment]
Thanks.

However, I was following your l[azyvim-starter-for-quarto](https://github.com/jmbuhr/lazyvim-starter-for-quarto/blob/main/lua/plugins/quarto.lua), and I don't know how to add the custom `on_attach_qmd` to the `marksman` related config table in my `nvim_lspconfig.lua` file: 

https://github.com/hopezh/oatLazyVim/blob/main/lua/plugins/nvim_lspconfig.lua

**Reply by jmbuhr** [reply]
I mean, I can't look through your whole config, you'll have to know what's in there. You can start by checking for the latest time the `K` keymap is set with https://neovim.io/doc/user/map.html#%3Amap-verbose

**Comment by hopezh** [comment, marked as answer]
Thanks, @jmbuhr , and of course, I'm not asking you to dig into my config file. 

Anyway, after some poking around, I managed to solve the issue by moving the line `--@param opts cmp.ConfigSchema` out of the `opts` function in the config of `nvim-cmp` in my [quarto.lua](https://github.com/hopezh/oatLazyVim/blob/d57d9bf4f8f107c65c956cb21743eb2fc1c238f3/lua/plugins/quarto.lua#L101C38-L101C38) file: 

<img width="1629" alt="Screenshot 2023-11-04 at 21 50 08" src="https://github.com/quarto-dev/quarto-nvim/assets/23435810/d15884b4-46a2-404f-b0be-c2e39f0efebc">

**Reply by jmbuhr** [reply]
Interesting, and very curious indeed since this is just a lua comment, albeit with some magic syntax for the lua language server, but it shouldn't change behaviour. Maybe lazy.nvim somehow uses it for some validation.

**Reply by hopezh** [reply]
...You're right, seems that the position of the `--@param` part is not relevant here...

I'm unable to replicate the "lsp missing" issue I posted here now...

Anyway, I'll pay attention to what may trigger this issue.
